Gui/ViewerGL.cpp

```diff
@@ -3617,7 +3631,11 @@ ViewerGL::swapOpenGLBuffers()
 {
     // always running in the main thread
     assert( qApp && qApp->thread() == QThread::currentThread() );
+#if QT_VERSION >= QT_VERSION_CHECK(5, 4, 0)
+    update();
```

Comment:
Can you explain this change? I don't understand it.

Reply:
ok, as I understand from the doc, no swapBuffers() in QOpenGLWidget, but shouldn't it call https://doc.qt.io/qt-5/qopenglcontext.html#swapBuffers instead?

Reply:
In Qt5+ it's preferable to call `QOpenGLWidget::update`, which invokes `glSwapBuffers` (mentioned in the comment and in the docs), over to `QOpenGLContext::swapBuffers` in a Qt Widgets scenario because you have to pass a `QSurface` (e.g. `QWindow`) to the latter, which it's way more harder to setup and access in a widget application instead of a Qt Gui application.